build.gradle

```diff
@@ -263,7 +263,7 @@ artifactory {
     publish {
         contextUrl = artifactory_contextUrl
         repository {
-            repoKey = 'corda-dev'
+            repoKey =  System.getenv('CORDA_PUBLISH_REPOSITORY_KEY') ?: System.getProperty('corda.artifactory.repokey', 'corda-dev')
             username = project.findProperty('cordaArtifactoryUsername') ?: System.getenv('CORDA_ARTIFACTORY_USERNAME') ?: System.getProperty('corda.artifactory.username')
             password = project.findProperty('cordaArtifactoryPassword') ?: System.getenv('CORDA_ARTIFACTORY_PASSWORD') ?: System.getProperty('corda.artifactory.password')
```

Comment:
Do you know what is using these Java system properties, please? Because I thought Jenkins only uses environment variables :thinking:.

Reply:
actually your right, Jenkins will set a [env var](https://github.com/corda/corda-shared-build-pipeline-steps/blob/ronanb/CORE-4377/snyk-scanning_/vars/cordaPipeline.groovy#L122).

I am not sure why this pattern of using java system properties is used historically, the following may be more appropriate.

```
            repoKey =  System.getenv('CORDA_PUBLISH_REPOSITORY_KEY') ?: project.findProperty('corda.artifactory.repokey', 'corda-dev')
            username = System.getenv('CORDA_ARTIFACTORY_USERNAME') ?: project.findProperty('corda.artifactory.username')
            password = System.getenv('CORDA_ARTIFACTORY_PASSWORD') ?: project.findProperty('corda.artifactory.password')
```

Reply:
I think we've conventionally looked for Gradle properties first and environment variables second so that people can override environment variables using Gradle properties.

I see _no point at all_ in having Java system properties here.

Reply:
Sure, Updated